pkg/vsphere/tasks/waiter.go

```diff
@@ -27,12 +28,40 @@ import (
 	"github.com/vmware/govmomi/vim25/progress"
 	"github.com/vmware/govmomi/vim25/soap"
 	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/pkg/trace"
 )
 
 const (
 	maxBackoffFactor = int64(16)
+
+	VMContextObjectKey = "VMContextObject"
+)
+
+var (
+	m             sync.RWMutex
+	errorHandlers []func(ctx context.Context, err error) (bool, error)
```

Comment:
> Also, does the handler need to have additional context (not the go package), i.e. additional arguments? Like maybe task info, or the object that was involved in the task?

Now we're leveraging context to keep key/value for handler required objects, that's used in fix_container handler. I would rather to keep that behavior.
In the future, if the error handler relies on task or taskinfo, we can add them into context. I didn't do that right now, because no handler need that at this time yet.